### PR TITLE
Fix incorrect syntax for config

### PR DESCRIPTION
**Pull request by joel-hamill**
Fixed improper syntax which renders improperly in the downstream output.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by joel-hamill** [comment]
cc @cherylws

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 1]
```diff
@@ -285,8 +285,10 @@ public class ProducerConfig extends AbstractConfig {
             "<ul>" +
             "<li>If not set, the default partitioning logic is used. " + 
             "This strategy send records to a partition until at least " + BATCH_SIZE_CONFIG + " bytes is produced to the partition. It works with the strategy:" + 
-            "<p> 1) If no partition is specified but a key is present, choose a partition based on a hash of the key." +
-            "<p> 2) If no partition or key is present, choose the sticky partition that changes when at least " + BATCH_SIZE_CONFIG + " bytes are produced to the partition." +
+            "<ol>" +
+            "<li>If no partition is specified but a key is present, choose a partition based on a hash of the key." +
```
Comment (mjsax) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]:
```suggestion
            "<li>If no partition is specified but a key is present, choose a partition based on a hash of the key.</li>" +
```

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 2]
```diff
@@ -285,8 +285,10 @@ public class ProducerConfig extends AbstractConfig {
...
+            "<li>If no partition or key is present, choose the sticky partition that changes when at least " + BATCH_SIZE_CONFIG + " bytes are produced to the partition." +
```
Comment (mjsax) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]:
```suggestion
            "<li>If no partition or key is present, choose the sticky partition that changes when at least " + BATCH_SIZE_CONFIG + " bytes are produced to the partition.</li>" +
```

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `3.7` and `3.6` branches.

**Pull request merged**